Nodejs/Tests/TestAdapter.Tests/TestProjectFactory.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+using static Microsoft.NodejsTools.TestAdapter.TestFrameworks.TestFramework;
+
+namespace TestAdapter.Tests
+{
+    public class TestProjectFactory
+    {
+        public enum ProjectName
+        {
+            NodeAppWithTestsConfiguredOnProject,
+            NodeAppWithTestsConfiguredPerFile,
+            NodeAppWithAngularTests,
+        }
+
+        private readonly ProjectName projectName;
+        private string projectNameString => Enum.GetName(typeof(ProjectName), this.projectName);
+
+        public TestProjectFactory(ProjectName projectName)
+        {
+            this.projectName = projectName;
+        }
+
+        public string GetProjectDirPath()
+        {
+            return Path.Combine(Directory.GetCurrentDirectory(), $@"..\..\..\..\MockProjects\{projectNameString}");
+        }
+
+        public string GetProjectFilePath()
+        {
+            switch (this.projectName)
+            {
+                case ProjectName.NodeAppWithTestsConfiguredOnProject:
+                case ProjectName.NodeAppWithTestsConfiguredPerFile:
+                case ProjectName.NodeAppWithAngularTests:
+                    return Path.Combine(this.GetProjectDirPath(), $"{projectNameString}.njsproj");
+            }
+
+            throw new ArgumentException();
```

Comment:
Add a message to this exception so it's clear what unexpected project name was observed?

Reply:
Added message and changed to `NotImplementedException` when a projectName has not been implemented instead.